addons/hr/static/tests/standalone_m2o_avatar_employee_tests.js

```diff
@@ -0,0 +1,118 @@
+odoo.define('hr.StandaloneM2OEmployeeTests', function (require) {
+    "use strict";
+
+    const {xml} = owl.tags;
+
+    const AbstractRendererOwl = require('web.AbstractRendererOwl');
+    const BasicView = require("web.BasicView");
+    const BasicRenderer = require("web.BasicRenderer");
+    const RendererWrapper = require('web.RendererWrapper');
+    const {createView} = require('web.test_utils');
+
+    const StandaloneM2OAvatarEmployee = require('hr.StandaloneM2OAvatarEmployee');
+
+    function getHtmlRenderer(html) {
+        return BasicRenderer.extend({
+            start: function () {
+                this.$el.html(html);
+                return this._super.apply(this, arguments);
+            }
+        });
+    }
+
+    function getOwlView(owlRenderer, viewType) {
+        viewType = viewType || "test";
+        return BasicView.extend({
+            viewType: viewType,
+            config: _.extend({}, BasicView.prototype.config, {
+                Renderer: owlRenderer,
+            }),
+            getRenderer() {
+                return new RendererWrapper(null, this.config.Renderer, {});
+            }
+        });
+    }
+
+    function getHtmlView(html, viewType) {
+        viewType = viewType || "test";
+        return BasicView.extend({
+            viewType: viewType,
+            config: _.extend({}, BasicView.prototype.config, {
+                Renderer: getHtmlRenderer(html)
+            })
+        });
+    }
+
+    QUnit.module('hr', {}, function () {
+        QUnit.module('StandaloneM2OEmployeeTests', {
+            beforeEach: function () {
+                this.data = {
+                    'foo': {
+                        fields: {
+                            employee_id: {string: "Employee", type: 'many2one', relation: 'hr.employee'},
+                        },
+                        records: [],
+                    },
+                    'hr.employee': {
+                        fields: {},
+                        records: [
+                            {id: 10, name: "Mario"},
+                            {id: 20, name: "Luigi"},
+                            {id: 30, name: "Yoshi"}
+                        ],
+                    },
+                };
+            },
+        });
+
+        QUnit.test('standalone_m2o_avatar_employee: legacy view', async function (assert) {
+            assert.expect(1);
+
+            const html = "<div class='coucou_test'></div>";
+            const view = await createView({
+                View: getHtmlView(html, "test"),
+                data: this.data,
+                model: "foo",
+                arch: "<test/>"
+            });
+
+            const avatar10 = new StandaloneM2OAvatarEmployee(view, 10);
+            const avatar20 = new StandaloneM2OAvatarEmployee(view, 20);
+            const avatar30 = new StandaloneM2OAvatarEmployee(view, [30, 'Bowser']);
+
+            await avatar10.appendTo(view.$('.coucou_test'));
```

Comment:
view.$ -->  view.el.querySelector

(pure js instead of jQuery)

Reply:
OK I'll keep that in mind :+1: 
About line 87,
```js
... view.$('.o_field_many2one_avatar span').text() ...
```
Is there a pure js alternative ?

Reply:
[...view.el.querySelectorAll('.o_field_many2one_avatar span')] give you an array with all the span. you can then use .map and join to get what you want or a deepEqual which is better

Reply:
see below